docs/Icons.md

```diff
@@ -103,11 +94,36 @@ In some rare cases, you may wish to pass a simple string (e.g., a single letter
 <ComponentName icon={'🍇'} />
 ```
 
-## Image Source
+### Image Source
 
 The icon prop will also support images in various formats, either from local resources or from the web.
 
 ```tsx
 <Component icon={{ uri: 'https://raw.githubusercontent.com/etn-ccis/blui-icons/dev/packages/png/png48/account_settings_black500_48dp.png' }} />
 <Component icon={require('./path/to/image.png')} />
 ```
+
+## Icon as a component
+
+`<Icon>` component is an internal utility component used to render icons inside other components. It standardizes the implementation of the icon and handles all of the different ways to specify the icon without having to duplicate this logic inside every component that utilizes icons.
+
+<img width="100%" style="max-width: 100px; display: block;" alt="Icon" src="./images/icon.png">
+
+### Usage
+
+```tsx
+import { Icon } from '@brightlayer-ui/react-native-components';
+
+<Icon source={{ family: 'brightlayer-ui', name: 'device' }} />;
+```
+
+### API
+
+<div style="overflow: auto;">
+
+| Prop Name | Description                                                     | Type                                   | Required | Default                   |
+| --------- | --------------------------------------------------------------- | -------------------------------------- | -------- | ------------------------- |
+| source    | An object has the image source or an icon family or JSX.Element | [`IconSource`](#icon-object)           | yes      |                           |
```

Comment:
you will want to 
* not write this a sentence, but a description of `source`
* make it succinct 

```suggestion
| source    | An object to specify the source of the icon | [`IconSource`](#icon-object)           | yes      |                           |
```